src/hooks/useAllActivities.ts

```diff
@@ -1,18 +1,20 @@
 import { useEffect, useState } from "react";
-import { baseURL } from "../config";
-import { ApiResponse, ActivityItem } from "../api/activities.api.types";
+import { ActivityItem } from "../api/activities.api.types";
 import { getAllActivities } from "../api/activities.api";
 
 export function useAllActivities() {
-    const [apiResponse, setApiResponse] = useState({ data: [] as ActivityItem[], error: undefined });
+    const [apiResponse, setApiResponse] = useState({ data: [] as ActivityItem[], error: undefined});
+    const [loading, setLoading] = useState<boolean>(false);
```

Comment:
again the variable could be named idLoading, setIsLoading since it is checking for a boolean state

Reply:
Fixed